README.rst

```diff
@@ -124,16 +124,20 @@ Depending on your OS and environment, some of these packages may be more challen
 or any issue regarding the above packages, please check their respective documentation.
 
 
-Having trouble with PySCF?
+Quantum Chemistry Packages
```

Comment:
We had another section just before that saying "optional dependencies".
Can we make things flow a bit better ?

It feels like we now have a optional dependency section with a first "sub section" about quantum circuit backends, and another one about classical quantum chemistry backends. Maybe those are the "sub titles" we need, we throw 1-2 links and tell people to look up the installation instructions of these things (e.g it's not our problem)